pact/Pact/Core/Errors.hs

```diff
@@ -1061,6 +1095,586 @@ deriveConstrInfo ''EvalError
+      , tInt actual]
+  -- Library decoding error: do not serialise
+  HyperlaneErrorInvalidChainId _ ->
+    thsep ["Failed to decode chainId"]
+
+hyperlaneDecodeErrorToBoundedText' :: HyperlaneDecodeError -> Text
+hyperlaneDecodeErrorToBoundedText' = \case
+  HyperlaneDecodeErrorBase64 -> "Failed to base64-decode token message"
+  -- Library error, do not serialise
+  HyperlaneDecodeErrorInternal _ ->
+    "Hyperlane Native Internal Decoding error"
+  HyperlaneDecodeErrorBinary -> "Decoding error: binary decoding failed"
+  HyperlaneDecodeErrorParseRecipient -> "Could not parse recipient into a guard"
+
+-- | NOTE: Do _not_ change this function post mainnet release just to improve an error.
+--  This will fork the chain, these messages will make it into outputs.
+--  See [Bounded Text Errors]
+evalErrorToBoundedText :: EvalError -> BoundedText PactErrorMsgSize
+evalErrorToBoundedText = mkBoundedText . \case
+  ArrayOutOfBoundsException len ix ->
+    thsep
+    [ "Array index out of bounds. Length"
+    , tInt len <> ","
+    , "Index"
+    , tInt ix]
+  ArithmeticException txt ->
+    thsep ["Arithmetic exception:", txt]
+  EnumerationError txt ->
+    thsep ["Enumeration error:", txt]
+  DecodeError txt ->
+    thsep ["Decoding error:", txt]
+  FloatingPointError txt ->
+    thsep ["Floating point error:", txt]
+  -- Todo: probably enhance this data type
+  GasExceeded ->
+    "Gas limit exceeded"
+  InvariantFailure msg ->
+    thsep
+      ["Execution Invariant error, please report this to the pact team @kadena:", invariantErrorToBoundedText' msg]
+  NativeArgumentsError (NativeName n) tys ->
+    thsep (["Native evaluation error for native", n <> ",", "received incorrect argument(s) of type(s)"] ++ (renderArgTypeError <$> tys))
+  EvalError txt ->
+    thsep ["Program encountered an unhandled raised error:", txt]
+  YieldOutsideDefPact ->
+    "Try to yield a value outside a running DefPact execution"
+  NoActiveDefPactExec ->
+    "No active DefPact execution in the environment"
+  NoYieldInDefPactStep (DefPactStep step _ (DefPactId dpid) _) ->
+    thsep ["No yield in DefPactStep.", "Step:", tInt step, "DefPactId:", dpid]
+  InvalidDefPactStepSupplied (DefPactStep step _ _ _) stepCount ->
+    thsep
+    [ "DefPactStep does not match DefPact properties:"
+    , "requested:", tInt step
+    , "step count:", tInt stepCount]
+  DefPactIdMismatch (DefPactId reqId) (DefPactId envId) ->
+    thsep
+    [ "Requested DefPactId:", reqId
+    , "does not match context DefPactId:", envId
+    ]
+  CCDefPactContinuationError pactStep _ccExec _dbExec ->
+    thsep
+    [ "Crosschain DefPact continuation error:"
+    , "DefPactId:", (tDpId (_psDefPactId pactStep))
+    ]
+  NestedDefPactParentRollbackMismatch pid rollback parentRollback ->
+    thsep
+    [ "Nested DefPact execution failed, parameter missmatch:"
+    , "DefPactId:", tDpId pid
+    , "Rollback:", tBool rollback
+    , "Parent rollback:", tBool parentRollback
```

Comment:
Is this a nested defpact thing? Can you explain the message to me or maybe make it clearer?